processing/src/main/java/org/apache/druid/segment/join/filter/JoinFilterAnalyzer.java

```diff
@@ -265,6 +286,10 @@ public static JoinFilterPreAnalysis computeJoinFilterPreAnalysis(
                   (rhsCol) -> Optional.of(new ArrayList<>())
               );
           perColumnCorrelations.get().add(correlationForPrefix.getValue());
```

Comment:
get() without isPresent() check

Reply:
I added an assert for `perColumnCorrelations.isPresent()` here, it should never be empty in that block (for it to have been empty, `if (correlationsForPrefix.isPresent()) {` must have been false, and this block ensures it's not empty in the `computeIfAbsent` call)

I think this area of the code is hard to follow right now, it would be an area to be refactored.